### DISCUSSION TITLE
[BUG] lightgbm.basic.booster classifier was recognized as regressor

### DISCUSSION THREAD

**Discussion by omrialo** [deepchecks/deepchecks#1838] [Q&A] [answered]
Hi,
My model, which is object of type `lightgbm.basic.Booster`, was recognized by deepchecks as a regressor, while it is a classifier.
When I used the method `model.dump_model()`, the objective was:
`'objective': 'binary sigmoid:1'`

Could you please look into that?

**Comment by noamzbr** [comment, marked as answer]
Hi @omrialo. The checks and suites in deepchecks expect the model to provide the methods of the sklearn api for models, as detailed in our [Supported Models guide](https://docs.deepchecks.com/stable/user-guide/tabular/supported_models.html). The way deepchecks recognizes classification models is by checking if they have a `predict_proba` method, which deepchecks uses to compute the predicted class probabilities for all classes. Models with only a `predict` method are assumed to be regression models.

In the case of `lightgbm.basic.Booster`, it has only a `predict` method, that either produces the class probability of the second class for binary classification, or of all classes for multiclass. The solution for that is to write a short model wrapper object that will make sure the model provides to deepchecks the api it expects. For example, in your case the wrapper can be:

```
import numpy as np

class ModelWrapper:
    
    def __init__(self, model):
        self.model = model
    
    def predict(self, data):
        return (self.model.predict(data) > 0.5) * 1
    
    def predict_proba(self, data):
        predicted_class_1 = self.model.predict(data).reshape((-1, 1))
        return np.concatenate([1-predicted_class_1, predicted_class_1], axis=1)
```

Alternatively, you can train your model using the lightgbm sklearn api. For example, the [lightgbm.LGBMClassifier](https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMClassifier.html) object already implements the sklearn api and will work out of the box with deepchecks.
